Tests/mocks/devicesettings/FrontPanelTextDisplayMock.h

```diff
@@ -5,7 +5,26 @@
 class FrontPanelTextDisplayMock : public device::FrontPanelTextDisplayImpl {
 public:
     virtual ~FrontPanelTextDisplayMock() = default;
+    MOCK_METHOD(device::FrontPanelTextDisplay&, getInstanceByName, (const std::string& name), (override));
+    MOCK_METHOD(device::FrontPanelTextDisplay&, getInstanceById, (int id), (override));
+    MOCK_METHOD(int,getCurrentTimeFormat, (), (const, override));
+    MOCK_METHOD(void,setTimeFormat, (const int iTimeFormat), (const, override));
+    MOCK_METHOD(int,getTextBrightness, (), (const, override));
+    MOCK_METHOD(void,setTextBrightness, (const int brightness), (const, override));
+    MOCK_METHOD(void,setText, (const std::string text), (const, override));
+    MOCK_METHOD(void,setMode, (const int mode), (const, override));
 
-    MOCK_METHOD(int, getCurrentTimeFormat, (), (override));
-    MOCK_METHOD(void, setTimeFormat, (const int iTimeFormat), (override));
+    //FrontPanelIndicator mock functions, as TextDisplay extends front panel indicator.
```

Comment:
what happens if you derive from FrontPanelIndicatorMock?
```
class FrontPanelTextDisplayMock
 : public device::FrontPanelTextDisplayImpl
 , public FrontPanelIndicatorMock
```

Reply:
For whatever reason doing it as:
```
class FrontPanelTextDisplayMock
 : public device::FrontPanelTextDisplayImpl
 , public FrontPanelIndicatorMock
```
does not work, as it doesn't recognize that the virtual functions are being properly overrided and throws an error. I imagine it has something to do with how the mock_method call works.

Reply:
it should work. did you remove FrontPanelIndicatorMock's MOCK_METHOD-s and clean rebuild?

Reply:
@npoltorapavlo

#pragma once

#include <gmock/gmock.h>

#include "FrontPanelIndiciatorMock.h"

class FrontPanelTextDisplayMock : public device::FrontPanelTextDisplayImpl, public FrontPanelIndicatorMock {
```
public:
    virtual ~FrontPanelTextDisplayMock() = default;
    MOCK_METHOD(device::FrontPanelTextDisplay&, getInstanceByName, (const std::string& name), (override));
    MOCK_METHOD(device::FrontPanelTextDisplay&, getInstanceById, (int id), (override));
    MOCK_METHOD(int,getCurrentTimeFormat, (), (const, override));
    MOCK_METHOD(void,setTimeFormat, (const int iTimeFormat), (const, override));
    MOCK_METHOD(int,getTextBrightness, (), (const, override));
    MOCK_METHOD(void,setTextBrightness, (const int brightness), (const, override));
    MOCK_METHOD(void,setText, (const std::string text), (const, override));
    MOCK_METHOD(void,setMode, (const int mode), (const, override));
};
```
Is what the function looks like when I swap it over, and when I run a clean build I get the 'following functions are pure within' error message.

Reply:
@hgfell683 you had errors because FrontPanelTextDisplayImpl inherited from FrontPanelIndicatorImpl . FrontPanelTextDisplayMock inherited FrontPanelIndicatorImpl twice. in fact, no inheritance is needed  . fixed in https://github.com/rdkcentral/rdkservices/pull/3741